Improve error message if observer class not found

During initialization, Fluo just throws the following exception if an Observer class is not found.  It would better to catch this exception and print a more helpful error message instructing users how to fix problem.

```
Exception in thread "main" java.lang.RuntimeException: java.lang.ClassNotFoundException: io.fluo.stress.trie.NodeObserver
	at io.fluo.core.client.FluoAdminImpl.updateSharedConfig(FluoAdminImpl.java:149)
	at io.fluo.core.client.FluoAdminImpl.initialize(FluoAdminImpl.java:103)
	at io.fluo.cluster.init.Init.main(Init.java:126)
Caused by: java.lang.ClassNotFoundException: io.fluo.stress.trie.NodeObserver
	at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:190)
	at io.fluo.core.client.FluoAdminImpl.addObserver(FluoAdminImpl.java:167)
	at io.fluo.core.client.FluoAdminImpl.updateSharedConfig(FluoAdminImpl.java:137)
```

--- a/modules/cluster/src/main/java/io/fluo/cluster/init/Init.java
+++ b/modules/cluster/src/main/java/io/fluo/cluster/init/Init.java
@@ -28,6 +28,7 @@ import io.fluo.api.client.FluoAdmin;
 import io.fluo.api.client.FluoAdmin.InitOpts;
 import io.fluo.api.client.FluoFactory;
 import io.fluo.api.config.FluoConfiguration;
+import io.fluo.api.exceptions.FluoException;
 import io.fluo.core.client.FluoAdminImpl;
 import org.slf4j.LoggerFactory;
 
@@ -132,8 +133,14 @@ public class Init {
     }
 
     System.out.println("Initializing Fluo instance using " + commandOpts.getFluoProps());
-    FluoAdmin admin = FluoFactory.newAdmin(config);
-    admin.initialize(initOpts);
+    try {
+      FluoAdmin admin = FluoFactory.newAdmin(config);
+      admin.initialize(initOpts);
+    } catch (FluoException e) {
+      System.out.println("Initialization failed due to the following exception:");
+      e.printStackTrace();
+      System.exit(-1);
+    }
     System.out.println("Initialization is complete.");
   }
 }

--- a/modules/core/src/main/java/io/fluo/core/client/FluoAdminImpl.java
+++ b/modules/core/src/main/java/io/fluo/core/client/FluoAdminImpl.java
@@ -27,6 +27,7 @@ import io.fluo.api.client.FluoAdmin;
 import io.fluo.api.config.FluoConfiguration;
 import io.fluo.api.config.ObserverConfiguration;
 import io.fluo.api.data.Column;
+import io.fluo.api.exceptions.FluoException;
 import io.fluo.api.observer.Observer;
 import io.fluo.api.observer.Observer.NotificationType;
 import io.fluo.api.observer.Observer.ObservedColumn;
@@ -121,44 +122,52 @@ public class FluoAdminImpl implements FluoAdmin {
 
   @Override
   public void updateSharedConfig() {
-    
-    try {
-      logger.info("Setting up observers using app config: {}", ConfigurationUtils.toString(config.subset(FluoConfiguration.APP_PREFIX)));
-      
-      Map<Column,ObserverConfiguration> colObservers = new HashMap<>();
-      Map<Column,ObserverConfiguration> weakObservers = new HashMap<>();
-      for (ObserverConfiguration observerConfig : config.getObserverConfig()) {
-        
-        Observer observer = Class.forName(observerConfig.getClassName()).asSubclass(Observer.class).newInstance();
-        
-        logger.info("Setting up observer {} using params {}.", observer.getClass().getSimpleName(), observerConfig.getParameters());
-        observer.init(new ObserverContext(config.subset(FluoConfiguration.APP_PREFIX), observerConfig.getParameters()));
-        
-        ObservedColumn observedCol = observer.getObservedColumn();
-        if (observedCol.getType() == NotificationType.STRONG)
-          colObservers.put(observedCol.getColumn(), observerConfig);
-        else
-          weakObservers.put(observedCol.getColumn(), observerConfig);
+
+    logger.info("Setting up observers using app config: {}", ConfigurationUtils.toString(config.subset(FluoConfiguration.APP_PREFIX)));
+
+    Map<Column,ObserverConfiguration> colObservers = new HashMap<>();
+    Map<Column,ObserverConfiguration> weakObservers = new HashMap<>();
+    for (ObserverConfiguration observerConfig : config.getObserverConfig()) {
+
+      Observer observer;
+      try {
+        observer = Class.forName(observerConfig.getClassName()).asSubclass(Observer.class).newInstance();
+      } catch (ClassNotFoundException e1) {
+        throw new FluoException("Observer class '" + observerConfig.getClassName() + "' was not found.  Check for class name misspellings or failure to include the observer jar.", e1);
+      } catch (InstantiationException | IllegalAccessException e2) {
+        throw new FluoException("Observer class '" + observerConfig.getClassName() + "' could not be created.", e2);
       }
 
-      Properties sharedProps = new Properties();
-      Iterator<String> iter = config.getKeys();
-      while (iter.hasNext()) {
-        String key = iter.next();
-        if (key.equals(FluoConfiguration.TRANSACTION_ROLLBACK_TIME_PROP)) {
-          sharedProps.setProperty(key, Long.toString(config.getLong(key)));
-        } else if (key.startsWith(FluoConfiguration.APP_PREFIX)){
-          sharedProps.setProperty(key, config.getProperty(key).toString());
-        }
+      logger.info("Setting up observer {} using params {}.", observer.getClass().getSimpleName(), observerConfig.getParameters());
+      try {
+        observer.init(new ObserverContext(config.subset(FluoConfiguration.APP_PREFIX), observerConfig.getParameters()));
+      } catch (Exception e) {
+        throw new FluoException("Observer '" + observerConfig.getClassName() + "' could not be initialized", e);
       }
-      
+
+      ObservedColumn observedCol = observer.getObservedColumn();
+      if (observedCol.getType() == NotificationType.STRONG)
+        colObservers.put(observedCol.getColumn(), observerConfig);
+      else
+        weakObservers.put(observedCol.getColumn(), observerConfig);
+    }
+
+    Properties sharedProps = new Properties();
+    Iterator<String> iter = config.getKeys();
+    while (iter.hasNext()) {
+      String key = iter.next();
+      if (key.equals(FluoConfiguration.TRANSACTION_ROLLBACK_TIME_PROP)) {
+        sharedProps.setProperty(key, Long.toString(config.getLong(key)));
+      } else if (key.startsWith(FluoConfiguration.APP_PREFIX)){
+        sharedProps.setProperty(key, config.getProperty(key).toString());
+      }
+    }
+
+    try {
       Operations.updateObservers(config, colObservers, weakObservers);
       Operations.updateSharedConfig(config, sharedProps);
-      
     } catch (Exception e) {
-      if (e instanceof RuntimeException)
-        throw (RuntimeException) e;
-      throw new RuntimeException(e);
+      throw new FluoException("Failed to update shared configuration in Zookeeper", e);
     }
   }
   

